app/src/org/commcare/logic/AndroidFormController.java

```diff
@@ -88,4 +86,14 @@ public FormAnalyticsHelper getFormAnalyticsHelper() {
     public FormDef getFormDef() {
         return mFormEntryController.getModel().getForm();
     }
+
+    // TODO: we should cache this
+    public String getSerializedFormIndex() {
+        try{
+            byte[] serializedFormIndex = SerializationUtil.serialize(getFormIndex());
+            return Base64.encodeToString(serializedFormIndex, Base64.DEFAULT);
```

Comment:
can the value for `Base64.DEFAULT` change with OS updates ? If so, I would like to pass a concrete encoding here like `Charset.forName("UTF-8")`

Reply:
That's a good point, I'm not sure, let's lean to the safe side (but I will take note and look into that later)